winpr/libwinpr/utils/ntlm.c

```diff
@@ -77,52 +73,51 @@ BYTE* NTOWFv1A(LPSTR Password, UINT32 PasswordLength, BYTE* NtHash)
  * EndDefine
  */
 
-BYTE* NTOWFv2W(LPWSTR Password, UINT32 PasswordLength, LPWSTR User,
-		UINT32 UserLength, LPWSTR Domain, UINT32 DomainLength, BYTE* NtHash)
+BOOL NTOWFv2W(LPWSTR Password, UINT32 PasswordLength, LPWSTR User,
+              UINT32 UserLength, LPWSTR Domain, UINT32 DomainLength, BYTE* NtHash)
 {
 	BYTE* buffer;
 	BYTE NtHashV1[16];
-	BYTE* result = NtHash;
+	BOOL result = FALSE;
 
-	if ((!User) || (!Password))
-		return NULL;
+	if ((!User) || (!Password) || (!NtHash))
+		return FALSE;
 
 	if (!NtHash && !(NtHash = (BYTE*) malloc(WINPR_MD4_DIGEST_LENGTH)))
```

Comment:
This line should be also dropped :-/ I scheduled covscan again to be sure that I do not introduce another defect and this has been found :-) I am going to update it...